### PR TITLE
Refactor and test add professionals

### PR DESCRIPTION
**Pull request by austinandrade**
# Description

I've decided to refactor the code a bit and accommodate some edge cases in creating medical professionals. We have a separate endpoint for creating medical professionals (/create_professionals), so I went ahead and refactored it to a much cleaner and RESTful POST to '/medical_professionals'. I also decided to utilize the facade pattern to separate the controller/facade business logic into services. 

- Controllers will now handle routing logic and status codes
- Facades will be CRUD routers and will receive controller params and route to appropriate services containing business logic 
- Services now receive CRUD commands and execute them in file

For example: 

1. GET /medical_professionals routes to 'medical_professionals#index'
2. Index orders facade's get_medical_professionals method
3. Facade get_medical_professionals routes to Service with business logic
4. Service executes orders and returns existing medical professional records

## Type of change

Please delete options that are not relevant.

- [x] Refactor
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

100% coverage in a top-down feature-tested approach. More edge cases will be brainstormed and accounted for before the public API is created so that user experience is prioritized.

- [x] Feature Tested

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/controllers/api/v1/medical_professionals_controller.rb [hunk 1]
```diff
@@ -9,4 +9,42 @@ def index
       render json: MhpSerializer.new(medical_professionals).serializable_hash
     end
   end
+
+  def create
+    return render status: :unauthorized if unauthorized
+
+    case params[:profession]
+    when 'doctor'
+      MedicalProfessionalsFacade.create_doctor_records(
```
Comment (ghost) [inline app/controllers/api/v1/medical_professionals_controller.rb]: Yesssss, love it.

File: app/services/medical_professionals/create_service.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  class << self
+    def create_doctor(doctor_params, insurances, specialties, profession)
+      new_doctor = Doctor.new(doctor_params)
+      if new_doctor.save
+        if insurances && specialties
+          add_insurance(new_doctor, insurances, profession)
+          add_specialties(new_doctor, specialties, profession)
+        elsif insurances
+          add_insurance(new_doctor, insurances, profession)
+        elsif specialties
```
Comment (ghost) [inline app/services/medical_professionals/create_service.rb]: I think FE wants insurance to be required for doctors. So we would not need to cover this scenario.

File: config/routes.rb [hunk 3]
```diff
@@ -2,8 +2,7 @@
   namespace :api do
     namespace :v1 do
       get "/medical_professionals", to: 'medical_professionals#index'
-
-      post "/add_professional", to: 'add_professional#create'
+      post "/medical_professionals", to: 'medical_professionals#create'
```
Comment (ghost) [inline config/routes.rb]: maybe make this singular since they're only adding one at a time.

File: app/services/medical_professionals/create_service.rb (hunk 2 above)
Comment (austinandrade) [reply app/services/medical_professionals/create_service.rb]: Ok cool. Would specialties be optional?

File: config/routes.rb (hunk 3 above)
Comment (austinandrade) [reply config/routes.rb]: I've updated these to be resources instead of being hand-rolled. In REST though, changing pluralization establishes a new endpoint altogether.

For example: 
*'/medical_professionals'*
1. 'medical_professionals#index' will handle grabbing all of the doctor/mhp resources.
2. 'medical_professionals#show' would be for grabbing a single doctor/mhp resource.
3. 'medical_professionals#create' would be for creating a single doctor/mhp resource.
4. 'medical_professionals#update' would be for updating a single doctor/mhp resource.
5. 'medical_professionals#delete' would be for deleting a single doctor/mhp resource.

To my understanding, REST is the ability to change the verb of a single endpoint and continuously perform CRUD on a DB's resources. 

Let me know if that makes sense :)

File: app/services/medical_professionals/create_service.rb (hunk 2 above)
Comment (ghost) [reply app/services/medical_professionals/create_service.rb]: Not sure, maybe ask at standup

File: config/routes.rb (hunk 3 above)
Comment (ghost) [reply config/routes.rb]: Ahhhh indeed. Yes, thank you. I blame morning brain 😵

**Review by ghost — APPROVED** [review]
